src/lein_git_down/git_wagon.clj

```diff
@@ -53,7 +58,8 @@
 
 (defmethod resolve-pom! :leiningen
   [[_ project]]
-  (io/file (lein/apply-task "pom" (project/read (str project)) [])))
+  (hooke/with-scope
```

Comment:
With the new test in place I'm able to see fail-then-pass by applying the locking and with-scope in `lein-jar` below, but could you provide more context as to why `hooke/with-scope` is needed here as well?

Feel free to link me to further reading rather than restate things explained elsewhere.

Reply:
Yep, so Hooke does an `alter-var-root` replacing the "hooked" function with one that wraps it with the "hooks". The [`with-scope`](https://github.com/technomancy/robert-hooke/blob/master/src/robert/hooke.clj#L84) macro allows you to only apply the hooks within the context of the provided body and then "finally" revert the var back to its original value. If we didn't do this, then the var would be changed for the entire JVM process (it would be "hooked" during the pom phase) and every other operation that leveraged the "hooked" function would use the altered version. Essentially, it allows us to run each Lein operation "sandboxed" from the others so that their side effects don't bleed over. Let me know if you want to discuss more.